### PR TITLE
[JENKINS-57893] Mask Kubernetes secrets bound to containers in build log

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-57893](https://issues.jenkins-ci.org/browse/JENKINS-57893)

- [X] support Pipeline builds (using  jenkinsci/workflow-step-api-plugin#42 & jenkinsci/workflow-cps-plugin#291)
- [ ] support freestyle builds; not sure if `ConsoleLogFilter` could be used for this purpose
- [X] mask pod-level secrets
- [X] mask container-level secrets
- [X] mask secrets defined in YAML

@Vlatombe

### CODE REVIEW

**Comment by jglick** [comment]
`testBadLabel` failure is solved in #490.

**Comment by Vlatombe** [comment]
Noting upgrade of `workflow-cps` to 2.70

**Comment by jglick** [comment]
> support freestyle builds

Now I recall why I did not even attempt this originally: there is not so much as a single functional test which demonstrates running a freestyle build against `KubernetesCloud`. Thus by @batmat’s [rule](https://twitter.com/bmathus/status/1138055156946354178) it has to be considered an unsupported use case.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/SecretsMasker.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                        }
+                    }
+                }
+                if (!secretContainerKeys.isEmpty()) {
+                    String containerName = container.getName();
+                    LOGGER.fine(() -> "looking for " + slave.getNamespace() + "/" + slave.getPodName() + "/" + containerName + " secrets named " + secretContainerKeys);
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                    Semaphore semaphore = new Semaphore(0);
+                    try (ExecWatch exec = slave.getKubernetesCloud().connect().pods().inNamespace(slave.getNamespace()).withName(slave.getPodName()).inContainer(containerName)
+                            .writingOutput(baos).writingError(System.err).writingErrorChannel(System.err)
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/SecretsMasker.java]: `System.err` leftover from early attempts?

**Comment by Vlatombe** [comment]
Some flake (same as https://github.com/jenkinsci/kubernetes-plugin/pull/496 I guesss)

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/SecretsMasker.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/SecretsMasker.java]: Not a leftover, but could be improved.

**Pull request merged**